### PR TITLE
Directional Ray Traced Shadows

### PR DESCRIPTION
**Pull request by anisunity**
Fixing various bugs (shader graph, reflections)
Implementing ray traced directional shadows

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2FRayTracedDirectionalShadows&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

### CODE REVIEW

**Comment by anisunity** [comment]
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2FRayTracedDirectionalShadows&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

File: com.unity.render-pipelines.core/ShaderLibrary/Sampling/Sampling.hlsl [hunk 1]
```diff
@@ -115,6 +115,14 @@ real2 SampleDiskUniform(real u1, real u2)
     return r * real2(cosPhi, sinPhi);
 }
 
+real3 SampleConeUniform(real u1, real u2, real cos_theta)
+{
+    float r0 = cos_theta + u1 * (1.0f - cos_theta);
+    float r = sqrt(max(0.0, 1.0 - r0 * r0));
+    float phi = TWO_PI * u2;
+    return float3(r * cos(phi), r * sin(phi), r0);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/ShaderLibrary/Sampling/Sampling.hlsl]: float3 => real3

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs [hunk 2]
```diff
@@ -629,7 +623,34 @@ static void DrawShadowMapContent(SerializedHDLight serialized, Editor owner)
...
+                        EditorGUILayout.PropertyField(serialized.serializedLightData.filterTracedShadow, s_Styles.filterTracedShadow);
+                        EditorGUILayout.PropertyField(serialized.serializedLightData.filterSizeTraced, s_Styles.filterSizeTraced);
+                        EditorGUI.indentLevel--;
+                    }
+                }
 
+                if (serialized.settings.lightType.enumValueIndex == (int)LightType.Directional)
+                {
+                    EditorGUILayout.PropertyField(serialized.serializedLightData.useRayTracedShadows, s_Styles.useRayTracedShadows);
+                    if(serialized.serializedLightData.useRayTracedShadows.boolValue)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: think to do if () and not if( 
:)

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/Shadows/RaytracingShadow.compute [hunk 3]
```diff
@@ -351,3 +355,71 @@ void RaytracingAreaShadowNewSample(uint2 groupThreadId : SV_GroupThreadID, uint2
...
+
+    // Generate the new sample (follwing values of the sequence)
+    float2 noiseValue = float2(0.0, 0.0);
+    noiseValue.x = GetRaytracingNoiseSample(globalSampleIndex, 0, scramblingValue.x);
+    noiseValue.y = GetRaytracingNoiseSample(globalSampleIndex, 1, scramblingValue.y);
+
+    // Create the local ortho basis
+    float3x3 localToWorld = GetLocalFrame(-lightData.forward);
+
+    float3 localDir = SampleConeUniform(noiseValue.x, noiseValue.y, cos(_DirectionalLightAngle * 3.14 / 180.0));
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/Shadows/RaytracingShadow.compute]: >* 3.14 / 180.0
Add a MACRO for conversoin to radian :), don't we have one ?

**Pull request merged**